src/puppetlabs/puppetdb/http/query.clj

```diff
@@ -244,8 +245,12 @@
 (defn post-req->query
   "Takes a POST body and parses the JSON to create a pdb query map"
   [req]
-  (with-open [reader (-> req :body clojure.java.io/reader)]
-    (json/parse-stream reader true)))
+  (-> (with-open [reader (-> req :body clojure.java.io/reader)]
+        (json/parse-stream reader true))
+      (update-in [:query] (fn [query]
```

Comment:
I hate that there are two methodologies, btw. It just makes it a question of taste, which is an icky place to be.